### PR TITLE
fix: add number support for matches

### PR DESCRIPTION
**Pull request by marcosvega91**
**What**:add number support for matches

**Why**: because of an issue with typescript configuration I have dropped the support in #848 

**How**:

**Checklist**:

- [ ] Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs)
- [ ] Tests
- [ ] Typescript definitions updated
- [X] Ready to be merged

### CODE REVIEW

File: types/tsconfig.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "extends": "../tsconfig.json"
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {"@testing-library/dom": ["."]}
+  }
```
Comment (nickmccurdy) [inline types/tsconfig.json:6]: These config changes seem unrelated, unless you're planning on pushing more files to this PR

**Review by nickmccurdy — CHANGES_REQUESTED** [review]

File: types/tsconfig.json (hunk 1 above)
Comment (marcosvega91) [reply types/tsconfig.json:6]: this config is useful because types tests are now using the types from `node_modules` instead of using types from their folder. I you made some changes to types the tests will fail without it because is used the alias `'@testing-library/dom'` to import types instead of relative path

**Review by nickmccurdy — APPROVED** [review]
